src/misc.c

```diff
@@ -282,6 +285,39 @@ static JSValue tjs_print(JSContext *ctx, JSValueConst this_val, int argc, JSValu
     return JS_UNDEFINED;
 }
 
+static JSValue tjs_prompt(JSContext *ctx, JSValueConst this_val, int argc, JSValueConst *argv) {
+    char *str;
+
+    const char *message = "";
+    const char *default_value = "";
+
+    if (argc > 0) {
+        message = JS_ToCString(ctx, argv[0]);
+        if (!message) {
+            return JS_EXCEPTION;
+        }
+    }
+
+    Replxx* replxx = replxx_init();
+    
+    if (argc > 1) {
+        default_value = JS_ToCString(ctx, argv[1]);
+        replxx_set_preload_buffer(replxx, default_value);
+    }
+
+    str = strdup(replxx_input(replxx, message));
```

Comment:
No need to dup the string, is there? `JS_NewString` will copy it anyway. Also, is anything else needed for freeing the return value of `replxx_input` ?

Reply:
Fixed!